### PR TITLE
Fix inner border radius

### LINKED ISSUE #1904: CornerRadius with border and fill results in white gaps
If I change the `ContentPresenter` in the template for `Button`' to have a 4 pixel `CornerRadius` I see the following:

image

You can see that the fill and the border don't line up properly. I think `BorderRenderHelper` needs to adjust the corner radius of the fill depending on the border thickness somehow? This happens on both D2D and Skia so I don't think it's a renderer problem.

cc: @Gillibald because you wrote the code.

### PR DESCRIPTION
**Pull request by Gillibald**
- What does the pull request do?
Fixes issue Fix #1904
- What is the current behavior?
The background doesn't look right because the inner radius isn't calculated properly.
- What is the updated/expected behavior with this PR?
Everything should line up and rendered properly.
- How was the solution implemented (if it's not obvious)?
The inner radius is calculated by the thickness of the border and the outer radius.

Note: Fixed some code style issues.

Fixes #1904

### CODE REVIEW

File: src/Avalonia.Controls/Utils/BorderRenderHelper.cs [hunk 1]
```diff
@@ -88,24 +88,79 @@ public void Render(DrawingContext context, Size size, Thickness borders, CornerR
...
-                var borderThickness = borders.Left;
-                var cornerRadius = (float)radii.TopLeft;
-                var rect = new Rect(size);
+                var borderThickness = borders.Top;
+                var cornerRadius = (float)Math.Max(0, radii.TopLeft - borderThickness);
 
                 if (background != null)
                 {
-                    context.FillRectangle(background, rect.Deflate(borders), cornerRadius);
+                    var innerRect = new Rect(new Point(borders.Left, borders.Top), new Point(size.Width - borders.Right, size.Height - borders.Bottom));
```
[the remark this answers is not in the document]
Comment (Gillibald) [reply src/Avalonia.Controls/Utils/BorderRenderHelper.cs]: Had this before. Sure will do.

**Comment by ahopper** [comment]
this seems to work without the extra rectangle
```
 if (background != null)
                {
                    context.FillRectangle(background, rect.Deflate(borders),(float)(cornerRadius-borderThickness/2));
                }
```

**Comment by Gillibald** [comment]
@ahopper Have a close look. It doesn't work without it. Add a CornerRadius of 4 to the template ContentPresenter and you will still see some sub pixel error.

**Comment by ahopper** [comment]
@Gillibald it looks ok here under the magnifier with a range of borderthickness and radius, but if you can see something it must be there.  Am I correct in thinking the radius of the border is actually the radius of the centerline of the border?

**Comment by Gillibald** [comment]
@ahopper Thanks for the comment

**Review by grokys — APPROVED** [review]
Thanks that seems to have fixed it!

**Comment by SkyZero1228** [comment]
How can I add the Border Radius in Button?

**Pull request merged**